src/EFCore.PG/Query/ExpressionVisitors/NpgsqlSqlTranslatingExpressionVisitor.cs

```diff
@@ -137,14 +141,19 @@ protected virtual Expression VisitEqualsAny([NotNull] SubQueryExpression express
             {
                 // Translate someArray.Length
                 if (subQueryModel.ResultOperators.First() is CountResultOperator)
-                    return Expression.ArrayLength(Visit(fromExpression));
+                {
+                    if (Visit(fromExpression) is Expression from)
```

Comment:
Why not just add a single check above to test that `fromExpression` isn't null, and return there?